EIPS/eip-7160.md

```diff
@@ -37,65 +37,74 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
 
 ```solidity
 /// @title EIP-721 Multi-Metdata Extension
-/// @dev See https://eips.ethereum.org/EIPS/eip-721
-///  Note: the ERC-165 identifier for this interface is 0x06e1bc5b.
-interface IERC721MultiMetadata /* is IERC721Metadata */ {
-  /// @dev This event emits when a token uri is pinned and is
-  ///  useful for indexing purposes.
-  event TokenUriPinned(uint256 indexed tokenId, uint256 indexed index, address indexed sender);
-
-  /// @dev This event emits when a token uri is unpinned and is
-  ///  useful for indexing purposes.
-  event TokenUriUnpinned(uint256 indexed tokenId, address indexed sender);
-
-  /// @notice Get all token uris associated with a particular token
-  /// @dev If a token uri is pinned, the index returned should be the index in the string array
-  /// @param tokenId The identifier for the nft
-  /// @return index An unisgned integer that specifies which uri is pinned for a token (or the default uri if unpinned)
-  /// @return uris A string array of all uris associated with a token
-  function tokenURIs(uint256 tokenId) external view returns (uint256 index, string[] memory uris);
-
-  /// @notice Pin a specific token uri for a particular token
-  /// @param tokenId The identifier of the nft
-  /// @param index The index in the string array returned from the `tokenURIs` function that should be pinned for the token
-  function pinTokenURI(uint256 tokenId, uint256 index) external;
-
-  /// @notice Unpin metadata for a particular token
-  /// @dev This should reset the token to the default uri
-  /// @param tokenId The identifier of the nft
-  function unpinTokenURI(uint256 tokenId) external;
-
-  /// @notice Check on-chain if a token id has a pinned uri or not
-  /// @dev Useful for on-chain mechanics
-  /// @param tokenId The identifier of the nft
-  /// @return pinned A bool specifying if a token has metadata pinned or not
-  function hasPinnedTokenURI(uint256 tokenId) external view returns (bool pinned);
+/// @dev The ERC-165 identifier for this interface is 0x06e1bc5b.
+interface IERC7160 {
+
+    /// @dev This event emits when a token uri is pinned and is
+    ///  useful for indexing purposes.
+    event TokenUriPinned(uint256 indexed tokenId, uint256 indexed index);
+
+    /// @dev This event emits when a token uri is unpinned and is
+    ///  useful for indexing purposes.
+    event TokenUriUnpinned(uint256 indexed tokenId);
+
+    /// @notice Get all token uris associated with a particular token
+    /// @dev If a token uri is pinned, the index returned SHOULD be the index in the string array
+    /// @dev This call MUST revert if the token does not exist
```

Comment:
Would it make sense to have a standard event for when the implementation should revert? Or should we use the one from ERC721?

Reply:
sorry I meant custom error